### PR TITLE
Copter: takeoff relaxes wpnav and pos control during spool up

### PR DESCRIPTION
**Pull request by rmackay9**
This is a partial fix to improve takeoffs in Auto and Guided mode for helicopters and large multirotors that have a long spool up time.  The underlying issue is that during takeoff the waypoint navigation controller is running while the motors are still spooling up.  This leads to the altitude target climbing before the vehicle can actually get off the ground.

The fix is to relax the navigation, position and attitude controllers (yaw angle and reset rate I-terms) until the motors have completed spooling up.  These calls are very close to what is used in other modes including Loiter's **AltHold_Landed_Pre_Takeoff** submode. 

Below is a screen shot from SITL of before and after.  Note that in the **before** image the altitude target (in green) climbs before "spool up" completes while in the **after** image the target climbs as "spool up" completes.
copter-smoother-takeoff

This still needs testing on a real vehicle.

This is an alternative to PR: https://github.com/ArduPilot/ardupilot/pull/15207 (Copter: fix tradheli guided mode takeoff bug)

### CODE REVIEW

**Comment by rmackay9** [comment]
@bnsgeyer, thanks for adding the commit.  I swear I was going to pull it in eventually but thanks for saving me the trouble!

**Comment by bnsgeyer** [comment]
@rmackay9  No problem.  I was testing it out and thought I would save you the trouble.  It allows tradheli users to arm in an armable mode and then set Auto or Guided modes and enable motor interlock.   Motors wont start until user either sets mission start in auto mode or takeoff guided mode.  Then rotors spool up and takeoff isn't initiated until rotor runup is complete.  I tested it in SITL.  @rywunder has also tested this in a real heli.

**Review by bnsgeyer — APPROVED** [review]

File: ArduCopter/takeoff.cpp
Comment (peterbarker) [inline ArduCopter/takeoff.cpp:38]: This comment is now out-of-date.

Should probably just remove it - given it wasn't heli-specific it was *already* wrong....

File: ArduCopter/takeoff.cpp
Comment (peterbarker) [inline ArduCopter/takeoff.cpp:174]: This could be simplified by inverting the check against throttle-unlimitted and putting the set-land-complete outside  the if/else construct.  Just a suggestion, the patch is probably clearer as-is.

Comment (rmackay9) [reply ArduCopter/takeoff.cpp:174]: txs for the review and good point.  I might leave it as-is because it's consistent with other places in the code.

**Comment by rmackay9** [comment]
Looks like this is failing travis for a real reason or at least we need to look into it.

>>>> FAILED STEP: test.Helicopter at Tue Oct 13 23:44:23 2020
Failure Summary:
  test.Helicopter:
    PosHoldTakeOff (Fly POSHOLD takeoff) (Took off prematurely (abs(101.000000)>100.000000)) (see /tmp/buildlogs/HeliCopter-PosHoldTakeOff.txt)

**Comment by rmackay9** [comment]
I've rebased on master which seems to have a fix from @peterbarker related to the takeoff error.

**Comment by peterbarker** [comment]
On Tue, 13 Oct 2020, Randy Mackay wrote:

> Looks like this is failing travis for a real reason or at least we need to look into it.
>
>                         FAILED STEP: test.Helicopter at Tue Oct 13 23:44:23 2020
>                         Failure Summary:
>                         test.Helicopter:
>                         PosHoldTakeOff (Fly POSHOLD takeoff) (Took off prematurely (abs(101.000000)>100.000000)) (see
>                         /tmp/buildlogs/HeliCopter-PosHoldTakeOff.txt)

I've tried loosening the check from 100mm to 1000mm on the basis that we 
might have drifted that far....

I've *no* idea why this started to fail all of a sudden.

File: ArduCopter/takeoff.cpp
Comment (rmackay9) [reply ArduCopter/takeoff.cpp]: Thanks, I've fixed the comment in a follow-up commit.

**Comment by bnsgeyer** [comment]
@rmackay9 I don't necessarily agree with just changing the altitude error requirement for the self test. I made a significant change in the auto armed logic. I'm concerned that it may have affected other flight modes. I will check this tonight.￼

**Comment by rmackay9** [comment]
@bnsgeyer, ok thanks.  I'll leave it to you and @peterbarker to discuss the autotest change that went in.

**Comment by WickedShell** [comment]
Flight tested this earlier this week, seemed to work nicely and matches what I'd expect a takeoff from arming in auto to look like.

**Comment by rmackay9** [comment]
@bnsgeyer and I discussed this a few days ago, and he pointed out an issue in which the vehicle's motors spin down (and stop) when switched from Stabilize->Guided and conversely the motors can be stopped in Guided and then suddenly spin up when the vehicle is switched to Stabilize.  This is actually an existing issue but this change makes it more likely for a user to bump into it so we thought we should try and resolve it before merging this.

**Comment by bnsgeyer** [comment]
@rmackay9  just pushed a commit to fix the issue with spooling down when switching out of auto or guided to stabilize and vice versa.  It is not fully baked, but I wanted you to have a look to see how I'm going about things.  Its pretty close but I need to do a little more testing and maybe talk about landing and pausing during a mission and what the behavior should be.

**Comment by rmackay9** [comment]
I'm going to merge the minimum change to resolve the issue with copters jumping into the air.  We will handle the tradheli issues in a follow up PR certainly before we begin Copter-4.1 beta testing.

**Comment by bnsgeyer** [comment]
@rmackay9 So what was the rush of getting this merged.  Did you plan to backport to 4.0?  With a little discussion, I thought we could have made this acceptable for heli and killed two birds with one stone.

**Comment by rmackay9** [comment]
@bnsgeyer,

It's just taking us too long to get the larger issue resolved.  So I counter the "two birds with one stone" with "one bird in the hand is worth two in the bush".  There's a discussion [here on the forums](https://discuss.ardupilot.org/t/violent-takeoff-in-auto-mode/61947/3) and one of the Partners brought it up as well during their onboarding call.

**Comment by bnsgeyer** [comment]
@rmackay9  touché.   Ok I will probably resurrect my original PR.  I will need to talk to you about expected behavior (disarming?) after Land or RTL when in auto mission when the land or RTL are not the last nav command (meaning there is intent to takeoff again).

**Pull request merged**
